core/src/test/scala/nl/vroste/zio/kinesis/client/zionative/NativeConsumerTest.scala

```diff
@@ -799,7 +800,46 @@ object NativeConsumerTest extends ZIOSpecDefault {
               equalTo((0 until nrRecords).toList)
             )
         }
-      }
+      },
+      test("regression: correctly switch from parent to child when parent is empty") {
+        val nrShards  = 1
+        val nrShardsAfterScaling = 2
+        val nrRecords = 10
+
+        withRandomStreamAndApplicationName(nrShards) { (streamName, applicationName) =>
+          def consume(nr: Int) =
+            Consumer
+              .shardedStream(
+                streamName,
+                applicationName,
+                Serde.asciiString,
+                emitDiagnostic = e => ZIO.logDebug(e.toString)
+              )
+              .flatMapPar(nrShards, 1) { case (shard @ _, shardStream, checkpointer @ _) =>
+                shardStream.map((_, checkpointer))
+              }
+              .take(nr.toLong)
+              .debug
+              .mapError[Either[Throwable, ShardLeaseLost.type]](Left(_))
+              .tap { case (r, checkpointer) => checkpointer.checkpointNow[Any](r) }
+              .catchAll {
+                case Right(_) =>
+                  ZStream.empty
+                case Left(e)  => ZStream.fail(e)
+              }
+              .map(_._1)
+
+          for {
+            _                 <- produceSampleRecords(streamName, nrRecords, aggregated = true)
+            _                 <- consume(nrRecords).runCollect
+            _                 <- scaleStream(streamName, nrShardsAfterScaling)
+            consumer          <- consume(1).runCollect.forkScoped
+            parentsCompleted   = getCheckpoints(applicationName).map(_.values.count(_ == "SHARD_END") == nrShards)
+            _                 <- parentsCompleted.delay(200.millis).repeatUntil(identity)
+            _                 <- consumer.interrupt
+          } yield assertCompletes
+        }
+      } @@ awsOnly,
```

Comment:
Doesn't work properly on localstack as consumption keeps failing with 'shard not active' after a scaling operation